src/panels/config/automation/sidebar/ha-automation-sidebar-card.ts

```diff
@@ -0,0 +1,179 @@
+import { mdiClose, mdiDotsVertical } from "@mdi/js";
+import { css, html, LitElement, nothing } from "lit";
+import { customElement, property } from "lit/decorators";
+import { classMap } from "lit/directives/class-map";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { stopPropagation } from "../../../../common/dom/stop_propagation";
+import "../../../../components/ha-card";
+import "../../../../components/ha-dialog-header";
+import "../../../../components/ha-icon-button";
+import "../../../../components/ha-md-button-menu";
+import "../../../../components/ha-md-divider";
+import "../../../../components/ha-md-menu-item";
+import type { HomeAssistant } from "../../../../types";
+import "../ha-automation-editor-warning";
+
+export interface SidebarOverflowMenuEntry {
+  clickAction: () => void;
+  disabled?: boolean;
+  label: string;
+  icon?: string;
+  danger?: boolean;
+}
+
+export type SidebarOverflowMenu = (SidebarOverflowMenuEntry | "separator")[];
+
+@customElement("ha-automation-sidebar-card")
+export default class HaAutomationSidebarCard extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ type: Boolean, attribute: "wide" }) public isWide = false;
+
+  @property({ type: Boolean, attribute: "yaml-mode" }) public yamlMode = false;
+
+  @property({ attribute: false }) public menuEntries: SidebarOverflowMenu = [];
+
+  @property({ attribute: false }) public warnings?: string[];
+
+  protected render() {
+    return html`
+      <ha-card
+        outlined
+        class=${classMap({
+          mobile: !this.isWide,
+          yaml: this.yamlMode,
+        })}
+      >
+        <ha-dialog-header>
+          <ha-icon-button
+            slot="navigationIcon"
+            .label=${this.hass.localize("ui.common.close")}
+            .path=${mdiClose}
+            @click=${this._closeSidebar}
+          ></ha-icon-button>
+          <slot slot="title" name="title"></slot>
+          <slot slot="subtitle" name="subtitle"></slot>
+          ${this.menuEntries.filter((entry) => entry !== "separator").length
+            ? html`
+                <ha-md-button-menu
+                  slot="actionItems"
+                  @click=${this._openOverflowMenu}
+                  @keydown=${stopPropagation}
+                  @closed=${stopPropagation}
+                  positioning="fixed"
+                >
+                  <ha-icon-button
+                    slot="trigger"
+                    .label=${this.hass.localize("ui.common.menu")}
+                    .path=${mdiDotsVertical}
+                  ></ha-icon-button>
+                  ${this.menuEntries.map((menuEntry) =>
```

Comment:
I would just make this a slot

Reply:
It could be the entire menu yeah, then you can even add a single icon button if you want.

But at least the menu items, I think, and either check if there is anything in the slot to render the menu, or add a attribute `has-menu`